services/device-registry-mongodb-quarkus/src/main/java/org/eclipse/hono/deviceregistry/mongodb/quarkus/KafkaMetricsSupportProducer.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.deviceregistry.mongodb.quarkus;
+
+import java.util.List;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.eclipse.hono.client.kafka.metrics.KafkaClientMetricsSupport;
+import org.eclipse.hono.client.kafka.metrics.KafkaMetricsOptions;
+import org.eclipse.hono.client.kafka.metrics.MicrometerKafkaClientMetricsSupport;
+import org.eclipse.hono.client.kafka.metrics.NoopKafkaClientMetricsSupport;
+import org.eclipse.hono.client.kafka.producer.MessagingKafkaProducerConfigProperties;
+import org.eclipse.hono.client.notification.kafka.NotificationKafkaProducerConfigProperties;
+
+import io.micrometer.core.instrument.MeterRegistry;
+
+/**
+ * Creates the metrics support to be used for Kafka clients from which metrics are to be fetched.
+ */
+@ApplicationScoped
+public class KafkaMetricsSupportProducer {
```

Comment:
this seems to be a verbatim copy of the class used for the JDBC registry. Can we may be re-use the same class for both registry variants in order to prevent code duplication?

Reply:
Well, where to put it then?
The `device-registry-base` module is currently independent of Quarkus (besides the `RegisterForReflection` annotation).
If we want to keep it that way, we would need a new `device-registry-base-quarkus` module.
But as we have dropped Spring Boot support anyway, I think we could move towards only one device-registry base module - also with Quarkus elements in there. This could also be applied to the adapter-base and service-base modules then, removing the quarkus-specific base modules there.
@sophokles73 WDYT?

Reply:
> But as we have dropped Spring Boot support anyway, I think we could move towards only one device-registry base module - also with Quarkus elements in there.

I agree.

> This could also be applied to the adapter-base and service-base modules then, removing the quarkus-specific base modules there.

Yes, but that should probably go into a separate PR ...

Reply:
Ok. I would move the above class as well as some other common code also in a separate PR then.